### PR TITLE
Eliminate dependence on feross/buffer internals

### PR DESCRIPTION
**Pull request by feross**
Based on PR #4 by @jscissr

### CODE REVIEW

**Comment by feross** [comment]
Released as 3.1.0.

File: index.js [hunk 1]
```diff
@@ -6,30 +6,20 @@
...
-  // then avoid a copy. Otherwise, create a `Buffer` with a copy.
-  var constructor = Buffer.TYPED_ARRAY_SUPPORT
-    ? function (arr) {
-      arr.__proto__ = Buffer.prototype
-      return arr
+module.exports = function typedarrayToBuffer (arr) {
+  if (isTypedArray(arr)) {
+    // To avoid a copy, use the typed array's underlying ArrayBuffer to back new Buffer
+    var buf = new Buffer(arr.buffer)
+    if (arr.byteOffset !== 0 || arr.byteLength !== arr.buffer.byteLength) {
```
Comment (jscissr) [inline index.js:16]: The first check is actually superfluous, because if both lengths are equal, there can't be an offset.

Comment (feross) [inline index.js:16]: Fair point.

Comment (feross) [inline index.js:16]: Fixed in 3.1.1.

**Comment by devongovett** [comment]
This breaks in old versions of node where `new Buffer(arrayBuffer)` does not work. In those versions of node, a copy is required. Let me know if you still want to support old versions of node, and I'll make a PR.

**Comment by feross** [comment]
@devongovett Node 0.10 will be completely unsupported (including no security fixes!) by Oct 2016, and 0.12 by Dec 2016. I'd rather just make folks who need 0.10 and 0.12 support depend on an older version of this package.

Though, to be fair, this probably should have been a semver major release. I didn't realize that it was dropping support for old versions of node.

**Pull request merged**